b2sdk/transfer/emerge/emerger.py

```diff
@@ -130,14 +158,26 @@ def emerge_stream(
 
         :param int min_part_size: lower limit of part size for the transfer planner, in bytes
         :param int max_part_size: upper limit of part size for the transfer planner, in bytes
-
+        :param str,None large_file_sha1: result file SHA1 hash or ``None`` if not known
         """
         planner = self.get_emerge_planner(
             min_part_size=min_part_size,
             recommended_upload_part_size=recommended_upload_part_size,
             max_part_size=max_part_size,
         )
+
+        # ensure that write_intent_iterator is an iterator (and not just iterable, like a list)
+        if not isinstance(write_intent_iterator, Iterator):
+            write_intent_iterator = iter(write_intent_iterator)
```

Comment:
emerge_stream is designed to accept iterables. You can't change the API by adding a strict requirement like that - it would break software for people who used that method. next() is the only guaranteed thing to work with the input for this thing.

Reply:
Agree, this `if` block is there to ensure backward compatibility.  

The docstring specifies that only **iterators** are accepted, so it should be objects with a `__next__()` method.

However, the method worked ok with **iterables** too (object with an `__iter__` method).  Passing a list of `WriteIntent` gave the expected result.  

I'm converting iterables to iterators if necessary, so they continue to both work.